util/network-devp2p/src/discovery.rs

```diff
@@ -171,6 +173,7 @@ impl<'a> Discovery<'a> {
 			discovery_id: NodeId::new(),
 			discovery_nodes: HashSet::new(),
 			node_buckets: (0..ADDRESS_BITS).map(|_| NodeBucket::new()).collect(),
+			other_observerd_nodes: LruCache::new(256),
```

Comment:
Do you think it's going to be enough? Maybe let's keep this number higher, I don't think it requires a lot of memory.

Reply:
It's just an _uneducated_ guess.
How many unrecognized nodes can ask us for peers in a real-life scenario?

Reply:
Hmmm, I think for our bootnodes it might be an order magnitude higher; I think something like 10000 should be enough (given in account a total number of live nodes in Ethereum mainnet =) And our records are about 80-90 bytes (64 bytes for NodeId, plus ipv6 address (16 bytes) and port (two bytes), plus a couple of bytes for the enum tracking and internal state keeping for the cache structure itself); so 10K records will end up in 1 megabyte of Ram in the worst case, which I'd say should be OK.

Thanks for bringing this up.